node/service/src/lib.rs

```diff
@@ -850,16 +870,56 @@ where
 		slot_duration_millis: slot_duration.as_millis() as u64,
 	};
 
+	// TODO: Do some manual tests for this.
```

Comment:
*Technically* at least under Linux I think it should be possible to automatically test this by e.g. 1) setting up a temporary directory tree according to what scenario we want to test, 2) calling `fork`, 3) in the fork `unshare`ing the user and the mount namespaces, 4) `chroot`/`pivot_root`ing to the directory tree we've set up, 5) running this snippet.

Would be nice to add a test for this just in case (because I could imagine this subtly breaking in the future by accident), but it's up to you.

Reply:
Agreed that this would be really nice to have. Unfortunately I am getting an "invalid argument" error when I run `unshare`, and I think it is because of this:

              CLONE_NEWUSER requires that the calling process is not
              threaded; specifying CLONE_NEWUSER automatically implies
              CLONE_THREAD.

Not sure if it makes a difference, but I'm not running `fork` first as you propose. I don't think it's sound in the Rust test environment[^0], so I've been trying [`rusty-fork`](https://github.com/AltSysrq/rusty-fork), but it doesn't help with the error.

[^0]: From `rusty-fork` docs: "The Rust test environment is multi-threaded, so attempting to do anything non-trivial after a process fork would result in undefined behaviour."

In lieue of that there is `fakechroot` for a rootless chroot, but it's not installed by default.

Another option to have the binary locations override-able by some test-only mutable globals. Each test would first get a lock on the globals for the duration of the test. Not sure what other options there are.

Reply:
> Another option to have the binary locations override-able by some test-only mutable globals.

I've implemented the tests using this option.